File: back-end/src/main/java/es/ull/project/application/usecase/operation/OperationCreateUseCase.java
package es.ull.project.application.usecase.operation;

import es.ull.project.domain.valueobject.Operation;
import es.ull.project.domain.valueobject.OperationName;
import es.ull.utils.docker.UllDockerImageName;

public interface OperationCreateUseCase {

    /**
     * Create an Operation
     * 
     * @param name
     * @param operationName
     * @param dockerName
     * @return operation created
     */
    Operation createOperation(OperationName name, UllDockerImageName dockerImage);
}

File: front-end/src/application/usecase/operation/operation-create-use-case.ts
import { Operation } from '../../../domain/value-object/operation';
import { OperationName } from '../../../domain/value-object/operation-name';
import { UllDockerImageName } from '@ull-tfg/ull-tfg-typescript/dist/ull-docker-image-name';
import { Either } from '@ull-tfg/ull-tfg-typescript';
import type { DataError } from '@ull-tfg/ull-tfg-typescript';
import { NodeIcon } from '@/domain/enumerate/node-icon';

export interface OperationCreateUseCase {
    /**
     * Create an Operation
     * @param name
     * @param dockerImage
     * @param icon
     * @returns operation created
     */
    createOperation(name: OperationName, dockerImage: UllDockerImageName): Promise<Either<DataError, Operation>>;
}
